pkg/parser/slurm.go

```diff
@@ -183,3 +198,185 @@ func GpusFlag(val string) (map[string]*resource.Quantity, error) {
 
 	return gpus, nil
 }
+
+func SlurmValidateAndReplaceScript(script string, nTasks int32) (string, error) {
+	if nTasks < 1 {
+		return "", fmt.Errorf("invalid number of tasks: %d", nTasks)
+	}
+
+	var (
+		sb          strings.Builder
+		srunCount   int
+		line        string
+		needNewLine bool
+	)
+
+	lines := strings.Split(script, "\n")
+
+	for i := 0; i < len(lines); i++ {
+		line = ""
+
+		// Convert multiline command to one line
+		for i < len(lines) {
```

Comment:
It's a bit hacky and difficult to follow to use the same index in 2 nested loops. Is it really necessary? What if we did something like:

```go
i := 0
for i < len(lines) {
    line := ...
    // build multiline command
    ...
    i++
}
```

Is that an option?

Reply:
Fixed. Thanks!